WordPress/src/main/java/org/wordpress/android/ui/mediapicker/MediaPickerViewModel.kt

```diff
@@ -323,15 +333,44 @@ class MediaPickerViewModel @Inject constructor(
     fun clickOnLastTappedIcon() = clickIcon(lastTappedIcon!!)
 
     private fun clickIcon(icon: MediaPickerIcon) {
-        if (icon == WP_STORIES_CAPTURE) {
+        if (icon is WpStoriesCapture) {
             if (!permissionsHandler.hasPermissionsToAccessPhotos()) {
                 _onPermissionsRequested.value = Event(PermissionsRequested.CAMERA)
                 lastTappedIcon = icon
                 return
             }
             AnalyticsTracker.track(MEDIA_PICKER_OPEN_WP_STORIES_CAPTURE)
         }
-        _onIconClicked.postValue(Event(IconClickEvent(icon, mediaPickerSetup.canMultiselect)))
+
+        // Do we need tracking here?; review tracking need.
+
+        _onIconClicked.postValue(Event(populateIconClickEvent(icon, mediaPickerSetup.canMultiselect)))
+    }
+
+    private fun populateIconClickEvent(icon: MediaPickerIcon, canMultiselect: Boolean): IconClickEvent {
+        val action: MediaPickerAction = when (icon) {
+            is ChooseFromAndroidDevice -> {
+                val allowedTypes = icon.allowedTypes
+                val (context, types) = when {
+                    listOf(IMAGE).containsAll(allowedTypes) -> {
```

Comment:
this is not super readable but I don't know if there is anything better :/. Maybe `allowedTypes.all { it == IMAGE }`? It's not great either.

Reply:
yep I was thinking to it also, but at the end could not find a clear winner; if it's fine with you I would keep it 🙇